include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/ArbitraryOrderFDTD.hpp

```diff
@@ -59,11 +62,17 @@ namespace picongpu
                         auto const term = (i % 2) ? -weights[i] : weights[i];
                         additionalFactor += term;
                     }
-                    if(SPEED_OF_LIGHT * SPEED_OF_LIGHT * DELTA_T * DELTA_T * INV_CELL2_SUM * additionalFactor
-                           * additionalFactor
-                       > 1.0_X)
+                    auto const invCorrectedCell2Sum = INV_CELL2_SUM * additionalFactor * additionalFactor;
+                    auto const maxC_DT = 1.0_X / math::sqrt(invCorrectedCell2Sum);
+                    if(SPEED_OF_LIGHT * SPEED_OF_LIGHT * DELTA_T * DELTA_T * invCorrectedCell2Sum > 1.0_X)
+                    {
                         throw std::runtime_error(
-                            "Courant-Friedrichs-Lewy condition check failed, check your grid.param file");
+                            std::string("Courant-Friedrichs-Lewy condition check failed, check your grid.param file\n")
+                            + "Courant Friedrichs Lewy condition: c * dt <= " + std::to_string(maxC_DT)
+                            + " ? (c * dt = " + std::to_string(SPEED_OF_LIGHT * DELTA_T) + ")");
```

Comment:
I think this is incorrect. From this code it appears `maxC_DT` is an upper bound for `(c * dt)^2`, not `c * dt`

Reply:
If so, it is the same issue for all checkers, not just this one

Reply:
No it is correct it should be `c * dt` this is equal to the output we already have.

Reply:
You are right, i got confused